### DISCUSSION TITLE
Query not cached

### DISCUSSION THREAD

**Discussion by hansyulian** [TanStack/query#8114] [Q&A]
I wonder why my query isn't getting cached. I have the following:

```

const [sortDirection, setSortDirection] = useState("asc");
  const { data } = useQuery({
    queryKey: [sortDirection],
    queryFn: () => {
      return fetch("/api/material?orderDirection=" + sortDirection);
    },    
  });
```

and when i toggle the sort direction, i can see that the request is being thrown again and again. is there something wrong?

**Comment by TkDodo** [comment]
impossible to say without seeing a reproduction

**Reply by hansyulian** [reply]
can use this: https://github.com/hansyulian/questions/tree/main/question-2024-10-01-tanstack-query-cache

**Reply by TkDodo** [reply]
expected behaviour since you haven’t set `staleTime`: 

- https://tanstack.com/query/latest/docs/framework/react/guides/important-defaults#important-defaults
- https://tanstack.com/query/latest/docs/framework/react/guides/caching
- https://tkdodo.eu/blog/practical-react-query#the-defaults-explained

**Reply by TkDodo** [reply]
oh and also this:
- https://tkdodo.eu/blog/react-query-as-a-state-manager
